### PR TITLE
fix: influxdb2 packages should depend on curl

### LINKED ISSUE #22225: InfluxDB packages should depend on curl
Influxdb debs and rpms include a wrapper script that's run by systemd, which uses `curl` to poll the API endpoint until it is ready. Although `curl` is a common tool, there could be a scenario where it is not present on the system when influxdb is installed. As such, the packages should list `curl` as a dependency so that it gets pulled in during installation.

- [x] [master](https://github.com/influxdata/influxdb/pull/22228)
- [x] [2.0](https://github.com/influxdata/influxdb/pull/22236)
- [x] [master-1.x](https://github.com/influxdata/influxdb/pull/22229)
- [x] [1.9](https://github.com/influxdata/influxdb/pull/22250)
- [x] [1.8](https://github.com/influxdata/influxdb/pull/22253)

### PR DESCRIPTION
**Pull request by codyshepherd**
Closes #22225 for `2.0`

The systemd wrapper script uses curl so it should be listed as a package dependency.

- [x] [CHANGELOG.md](https://github.com/influxdata/influxdb/blob/master/CHANGELOG.md) updated with a link to the PR (not the Issue)
- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [x] Rebased/mergeable
- [x] Tests pass

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -49,6 +49,7 @@ Because of the version bump to `go`, the macOS build for this release requires a
 1. [21977](https://github.com/influxdata/influxdb/pull/21977): Flux metaqueries for `_field` take fast path if `_measurement` is the only predicate.
 1. [22060](https://github.com/influxdata/influxdb/pull/22060): Copy names from mmapped memory before closing iterator
 1. [22052](https://github.com/influxdata/influxdb/pull/22052): systemd service -- handle 40x and block indefinitely
+1. [22236](https://github.com/influxdata/influxdb/pull/22236): fix: influxdb-server packages should depend on curl
```
Comment (danxmoran) [inline CHANGELOG.md:52]: Making it match `master`:
```suggestion
1. [22236](https://github.com/influxdata/influxdb/pull/22236): influxdb2 packages should depend on curl
```

Comment (codyshepherd) [reply CHANGELOG.md:52]: lol crap i thought i fixed that

File: CHANGELOG.md [hunk 2]
```diff
@@ -49,6 +49,7 @@ Because of the version bump to `go`, the macOS build for this release requires a
 1. [21977](https://github.com/influxdata/influxdb/pull/21977): Flux metaqueries for `_field` take fast path if `_measurement` is the only predicate.
 1. [22060](https://github.com/influxdata/influxdb/pull/22060): Copy names from mmapped memory before closing iterator
 1. [22052](https://github.com/influxdata/influxdb/pull/22052): systemd service -- handle 40x and block indefinitely
+1. [22236](https://github.com/influxdata/influxdb/pull/22236): influxdb2 packages should depend on curl
```
Comment (danxmoran) [inline CHANGELOG.md:52]: Agh sorry, this also needs to get bumped up under the `2.0.9` section I think

Comment (codyshepherd) [reply CHANGELOG.md:52]: ah true

**Review by danxmoran — APPROVED** [review]

**Pull request merged**